ecosystem/nft-metadata-crawler/nft-metadata-crawler-parser/src/utils/json_parser.rs

```diff
@@ -9,7 +18,59 @@ impl JSONParser {
      * Parses JSON from input URI.
      * Returns the underlying raw image URI, raw animation URI, and JSON.
      */
-    pub async fn parse(_uri: String) -> (Option<String>, Option<String>, Option<Value>) {
-        todo!();
+    pub async fn parse(uri: String) -> (Option<String>, Option<String>, Option<Value>) {
+        match Self::parse_json(uri.clone()).await {
+            Ok(out) => out,
+            Err(e) => {
+                error!("Error parsing JSON: {}", e);
+                (None, None, None)
+            },
+        }
+    }
+
+    /**
+     * Parses JSON from input URI.
+     */
+    async fn parse_json(
+        uri: String,
+    ) -> anyhow::Result<(Option<String>, Option<String>, Option<Value>)> {
+        let (mime, size) = get_uri_metadata(uri.clone()).await?;
+        if ImageFormat::from_mime_type(mime).is_some() {
+            error!("JSON parser received image URI, skipping");
+            return Ok((None, None, None));
+        } else if size > MAX_FILE_SIZE_BYTES {
+            error!("JSON parser received large file, skipping");
+            return Ok((None, None, None));
+        }
+
+        let op = || {
+            async {
+                info!("Sending request for token_uri {}", uri);
+
+                let result = reqwest::get(&uri).await.map_err(Error::transient)?;
```

Comment:
do you want to distinguish the error codes? e.g. retry on 404 would likely fail again, right?

Reply:
planning on adding this as optimization in the future